test/e2e/storage/persistent_volumes-local.go

```diff
@@ -87,62 +95,257 @@ var _ = framework.KubeDescribe("[Volume] PersistentVolumes-local [Feature:LocalP
 		}
+
+		// Choose the first node
+		node0 = &config.nodes[0]
 	})
 
 	Context("when one pod requests one prebound PVC", func() {
 		var (
 			testVol *localTestVolume
-			node    *v1.Node
 		)
 
 		BeforeEach(func() {
-			// Choose the first node
-			node = &config.nodes[0]
+			testVol = setupLocalVolumePVCPV(config, node0)
 		})
 
 		AfterEach(func() {
 			cleanupLocalVolume(config, testVol)
 			testVol = nil
 		})
 
-		It("should be able to mount and read from the volume", func() {
-			By("Initializing test volume")
-			testVol = setupLocalVolume(config, node)
+		It("should be able to mount and read from the volume using one-command containers", func() {
+			By("Creating a pod to read from the PV")
+			//testFileContent was written during setupLocalVolume
+			_, readCmd := createWriteAndReadCmds(volumeDir, testFile, "" /*writeTestFileContent*/)
+			podSpec := makeLocalPod(config, testVol, readCmd)
+			f.TestContainerOutput("pod reads PV", podSpec, 0, []string{testFileContent})
+		})
+
+		It("should be able to mount and write to the volume using one-command containers", func() {
+			By("Creating a pod to write to the PV")
+			writeCmd, readCmd := createWriteAndReadCmds(volumeDir, testFile, testVol.hostDir /*writeTestFileContent*/)
+			writeThenReadCmd := fmt.Sprintf("%s;%s", writeCmd, readCmd)
+			podSpec := makeLocalPod(config, testVol, writeThenReadCmd)
+			f.TestContainerOutput("pod writes to PV", podSpec, 0, []string{testVol.hostDir})
+		})
+	})
+
+	Context("when two pods request one prebound PVC serially", func() {
+		var (
+			testVol *localTestVolume
+		)
+
+		BeforeEach(func() {
+			testVol = setupLocalVolumePVCPV(config, node0)
+		})
+
+		AfterEach(func() {
+			cleanupLocalVolume(config, testVol)
+			testVol = nil
+		})
 
-			By("Creating local PVC and PV")
-			createLocalPVCPV(config, testVol)
+		It("should be able to mount volume, write from pod1, and read from pod2 using one-command containers", func() {
+			By("Creating pod1 to write to the PV")
+			writeCmd, readCmd := createWriteAndReadCmds(volumeDir, testFile, testVol.hostDir /*writeTestFileContent*/)
+			writeThenReadCmd := fmt.Sprintf("%s;%s", writeCmd, readCmd)
+			podSpec1 := makeLocalPod(config, testVol, writeThenReadCmd)
+			f.TestContainerOutput("pod writes to PV", podSpec1, 0, []string{testVol.hostDir})
 
-			By("Creating a pod to consume the PV")
-			readCmd := fmt.Sprintf("cat /mnt/volume1/%s", testFile)
-			podSpec := createLocalPod(config, testVol, readCmd)
-			f.TestContainerOutput("pod consumes PV", podSpec, 0, []string{testContents})
+			By("Creating pod2 to read from the PV")
+			podSpec2 := makeLocalPod(config, testVol, readCmd)
+			f.TestContainerOutput("pod reads PV", podSpec2, 0, []string{testVol.hostDir})
 		})
+	})
 
-		It("should be able to mount and write to the volume", func() {
-			By("Initializing test volume")
-			testVol = setupLocalVolume(config, node)
+	Context("one pod with one prebound PVC writes to PV", func() {
```

Comment:
I think this can fit under the first context since it's also just testing one pod